pkg/kubernetes/integration_test.go

```diff
@@ -0,0 +1,308 @@
+//go:build integration
+// +build integration
+
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/grafana/xk6-disruptor/pkg/kubernetes/helpers"
+	"github.com/grafana/xk6-disruptor/pkg/testutils/e2e/fixtures"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/k3s"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func createRandomTestNamespace(k8s Kubernetes) (string, error) {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-",
+		},
+	}
+
+	ns, err := k8s.Client().CoreV1().Namespaces().Create(context.TODO(), ns, metav1.CreateOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to create test namespace %w", err)
+	}
+
+	return ns.Name, nil
+}
+
+// regexMatcher process lines from logs and notifies when a match is found
+type regexMatcher struct {
+	exp   *regexp.Regexp
+	found chan (bool)
+}
+
+// Accept implements the LogConsumer interface
+func (a *regexMatcher) Accept(log testcontainers.Log) {
+	if a.exp.MatchString(string(log.Content)) {
+		a.found <- true
+	}
+}
+
+// waitForRegex waits until a match for the given regex is found in the log produced by the container
+func waitForRegex(ctx context.Context, container testcontainers.Container, exp string, timeout time.Duration) error {
```

Comment:
If we're passing a context here, couldn't we get rid of the `timeout` parameter and add it as a deadline to the context?

Reply:
It could be possible, and I don't oppose doing this change, but I think it obscures the API of the function. I mean that having the timeout as an argument makes it harder for someone to forget about setting a timeout in the context (or example by passing a `context.Background()`). Besides, the function is called `wait` so I think it also is natural to have such argument.